pandas/tests/frame/test_reshape.py

```diff
@@ -133,6 +133,24 @@ def test_stack_unstack(self):
         assert_frame_equal(unstacked_cols.T, df)
         assert_frame_equal(unstacked_cols_df['bar'].T, df)
 
+    def test_stack_mixed_level(self):
+        # GH 18310
+        levels = [range(3), [3, 'a'], [1, 2]]
+        # flat columns:
+        df = DataFrame(1, index=levels[0], columns=levels[1])
+        result = df.stack()
+        expected = Series(1, index=MultiIndex.from_product(levels[:2]))
+        assert_series_equal(result, expected)
+
+        # MultiIndex columns:
+        df = DataFrame(1, index=levels[0],
+                       columns=MultiIndex.from_product(levels[1:]))
+        result = df.stack(1)
```

Comment:
can you add the 2nd example you had in the issue as well, e.g. ``df[['a', 'b']].stack(0)`` (though different starting frame as well i think.

Reply:
(done)
ping